src/pygments_djc/lexers.py

```diff
@@ -0,0 +1,117 @@
+from pygments.lexer import Lexer, bygroups, using
+from pygments.lexers.css import CssLexer
+from pygments.lexers.javascript import JavascriptLexer
+from pygments.lexers.python import PythonLexer
+from pygments.lexers.templates import HtmlDjangoLexer
+from pygments.token import Name, Text, Operator, String, Punctuation
+
+
+# Since this Lexer will be used only in documentation, it's a naive implementation
+# that detects the nested JS / CSS / HTML blocks by searching for following patterns:
+# - `js = """`
+# - `js: some.Type = """`
+# - `css = """`
+# - `css: some.Type = """`
+# - `template = """`
+# - `template: some.Type = """`
+#
+# However, this Lexer is NOT sensitive to where the variable is defined. So this highlighting rule
+# will be triggered even when the variable is NOT defined on the Component class.
+#
+# In other words, we want to highlight cases like this:
+# ```py
+# class MyComponent(Component):
+#     template = """
+#     <div>Hello World</div>
+#     """
+# ```
+#
+# But NOT cases like this:
+# ```py
+# js = """
+# ```
+#
+# But our implementation still highlights the latter.
+def _gen_code_block_capture_rule(var_name: str, next_state: str) -> tuple[str, str, str]:
+    # In Pygments, capture rules are defined as a tuple of 3 elements:
+    # - The pattern to capture
+    # - The token to highlight
+    # - The next state
+    return (
```

Comment:
Here you can see that the returned value is a tuple of 3:
- The pattern to capture
- The token to highlight
- The next state (if any)